### PR TITLE
Feat e2e convert to pw suite coinmarket

### PR DESCRIPTION
**Pull request by Vere-Grey**
## Description
Feat e2e convert to pw suite coinmarket - buy-coins.test.ts

Keeps the mock data of the invity, which needs to be fixed
minor refacoring of fixture constructors

## Related Issue

CY to PW Migration

### CODE REVIEW

File: packages/suite-desktop-core/e2e/fixtures/invity/buy/info.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/fixtures/invity/buy/info.json:1]: These are the mock request responses. Currently not working.
I wonder whether I should keep them here or ditch them completely? What do you think?

File: packages/suite-desktop-core/e2e/snapshots/desktop/coin-market/buy-coins.test.ts/best-offer-buy-confirmation.png
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/snapshots/desktop/coin-market/buy-coins.test.ts/best-offer-buy-confirmation.png:1]: I see that I am having too narrow locator here -> I will fix this

File: packages/suite-desktop-core/e2e/playwright.config.ts [hunk 2]
```diff
@@ -40,6 +40,12 @@ const config: PlaywrightTestConfig = {
         : [['list'], ['html', { open: 'never' }]],
     timeout: process.env.GITHUB_ACTION ? timeoutCIRun : timeoutLocalRun,
     outputDir: path.join(__dirname, 'test-results'),
+    snapshotPathTemplate: 'snapshots/{projectName}/{testFilePath}/{arg}{ext}',
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/playwright.config.ts:45]: By default snapshots are stored in folder with test file. Which can lead to cluttered test folders. So this configuration puts snapshot to their own Folder.
On the other hand I am not sure how easy will be to find corresponding snapshot here. We will see and improve based on our experience

File: packages/suite-desktop-core/e2e/snapshots/desktop/coin-market/buy-coins.test.ts/best-offer-buy-confirmation.png
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/snapshots/desktop/coin-market/buy-coins.test.ts/best-offer-buy-confirmation.png:1]: I am not sure whether the receiving address will stay same all the time :/

File: packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        );
+    }
+
+    openCoinMarket = async () => {
+        await this.accountMenuButton.click();
+        await this.coinMarketBuyButton.click();
+        await this.waitForOffers();
+    };
+
+    waitForOffers = async () => {
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts]: Since we dont mock it. There is a chance this will fail because of invity failure. Lets monitor it and see if this will be   instability or not.
We might solve the invity mock soon and the risk will be gone

**Review by vdovhanych — COMMENTED** [review]
Snapshots are part of git for comparing it with the test run?

File: .github/workflows/test-suite-web-e2e-pw.yml [hunk 4]
```diff
@@ -32,6 +32,8 @@ on:
       - ".github/workflows/release*"
       - ".github/workflows/template*"
       - ".github/actions/release*/**"
+  schedule:
+    - cron: "0 0 * * *"
```
Comment (Vere-Grey) [inline .github/workflows/test-suite-web-e2e-pw.yml:36]: Configures the Web tests to be run as nightly. Copied from desktop pipeline

File: packages/suite-desktop-core/e2e/playwright.config.ts [hunk 5]
```diff
@@ -27,19 +27,25 @@ const config: PlaywrightTestConfig = {
     testDir: 'tests',
     workers: 1, // to disable parallelism between test files
     use: {
+        viewport: { width: 1280, height: 720 },
         headless: process.env.HEADLESS === 'true',
         trace: 'on',
         video: 'on',
         screenshot: 'on',
         testIdAttribute: 'data-testid',
-        actionTimeout: 30000,
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/playwright.config.ts:35]: I had some issue with traces and fails. Once I removed this forced timeout, it got fixed. Every action has its default timeout. So I suggest lets leave it to carefully chosen defaults by the playwright devs.

File: packages/suite-desktop-core/e2e/support/common.ts [hunk 6]
```diff
@@ -105,10 +105,15 @@ export const launchSuite = async (params: LaunchSuiteParams = {}) => {
     return { electronApp, window };
 };
 
+export const isDesktopProject = (testInfo: TestInfo) =>
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/support/common.ts:108]: I originally wanted to condition the visual comparison by this boolean. but then I found out you can add the --ignore-snapshot to project definition. But anyway I used this new methods on few places so I would keep it as a minor unrelated refactoring.

File: packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+            '@coinmarket/form/info/crypto-amount',
+        );
+        this.tradeConfirmationProvider = this.page.getByTestId('@coinmarket/form/info/provider');
+        this.tradeConfirmationContinueButton = this.page.getByTestId(
+            '@coinmarket/offer/continue-transaction-button',
+        );
+    }
+
+    openCoinMarket = async () => {
+        const walletPage = new WalletActions(this.page);
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts]: I feel this is just a little bit cleaner then getting the fixture walletPage here thru constructor.
What do you think?
:thinking:

Comment (Vere-Grey) [reply packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts]: Or should this method be completely in walletPage object and I should import that in test? Let me see locally

Comment (Vere-Grey) [reply packages/suite-desktop-core/e2e/support/pageActions/marketActions.ts]: Moved to WalletActions.

File: packages/suite-desktop-core/e2e/tests/settings/t2b1-device-settings.test.ts [hunk 8]
```diff
@@ -10,8 +10,9 @@ test.describe.serial('T2B1 - Device settings', { tag: ['@group=settings'] }, ()
         emulatorStartConf: { version: '2-latest', model: 'T2B1', wipe: true },
     });
 
-    test.beforeEach(async ({ onboardingPage, settingsPage }) => {
+    test.beforeEach(async ({ onboardingPage, dashboardPage, settingsPage }) => {
         await onboardingPage.completeOnboarding();
+        await dashboardPage.discoveryShouldFinish();
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/tests/settings/t2b1-device-settings.test.ts:15]: I had a random flaky fail on the run. I realized that the test is trying to manipulate with a setting that is dependent on the device discovery

File: packages/suite-desktop-core/e2e/tests/settings/t2t1-device-settings.test.ts [hunk 9]
```diff
@@ -2,8 +2,9 @@ import { test, expect } from '../../support/fixtures';
 
 test.describe('T2T1 - Device settings', { tag: ['@group=settings'] }, () => {
     test.use({ emulatorStartConf: { wipe: true, model: 'T2T1' } });
-    test.beforeEach(async ({ onboardingPage, settingsPage }) => {
+    test.beforeEach(async ({ onboardingPage, dashboardPage, settingsPage }) => {
         await onboardingPage.completeOnboarding();
+        await dashboardPage.discoveryShouldFinish();
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/tests/settings/t2t1-device-settings.test.ts:7]: I had a random flaky fail on the run. I realized that the test is trying to manipulate with a setting that is dependent on the device discovery

**Comment by Vere-Grey** [comment]
> Snapshots are part of git for comparing it with the test run?

Yeah, it is another tool next to asserts. It can both verify layout, UX and values. You create a snapshot, set how much it can differ, what to mask and then use it for comparison in following test runs.

There are options on hosting them apart from the repo. But I don't know much about that. I will read up on that to see what is the recommended approach.

**Review by HajekOndrej — APPROVED** [review]
